### PR TITLE
improved explanation of linear interpolation in quantile

### LINKED ISSUE #51745: improve explanation of linear interpolation in `quantile`
Currently explained like
https://github.com/pandas-dev/pandas/blob/2e218d10984e9919f0296931d92ea851c6a6faf5/pandas/core/series.py#L2647-L2648

The phrase "the fractional part of the index surrounded by `i` and `j`" is not easy to parse and has no clear meaning.

### PR DESCRIPTION
**Pull request by Detoro**
- [ ] closes #51745

### CODE REVIEW

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -1373,6 +1373,25 @@ class DateOffset(RelativeDeltaOffset, metaclass=OffsetMeta):
         previous midnight.
     **kwds
         Temporal parameter that add to or replace the offset value.
+    weekday : int {0, 1, ..., 6}, default 0
```
Comment (mroeschke) [inline pandas/_libs/tslibs/offsets.pyx:1376]: This shouldn't be here

File: pandas/core/series.py [hunk 2]
```diff
@@ -2683,13 +2683,18 @@ def quantile(
...
+                * linear: `i + (j - i) * fraction`, where `fraction` is the proportion
+                of the distance between `i` and `j`. it refers to the relative position
+                of the desired quantile value between i and j
+                hence fraction = (desired_quantile - i) / (j - i)
                 * lower: `i`.
                 * higher: `j`.
                 * nearest: `i` or `j` whichever is nearest.
                 * midpoint: (`i` + `j`) / 2.
 
+            For example, if (len(self) - 1) * q is 9.6, and the elements
```
Comment (mroeschke) [inline pandas/core/series.py:2695]: Can you add this as a code example in the `Examples` section instead?

Comment (Detoro) [reply pandas/core/series.py:2695]: Do I add it the same way it currently is as an explanation?

Comment (mroeschke) [reply pandas/core/series.py:2695]: You can create a `Series` with data that follows this example

Comment (Detoro) [reply pandas/core/series.py:2695]: Alright. Thank you

**Comment by mroeschke** [comment]
Closing in favor of https://github.com/pandas-dev/pandas/pull/54642, please push to the same branch next time

**Pull request closed without merging**